script/check-simulation-build-autotools

```diff
@@ -1,278 +0,0 @@
-
-set -euxo pipefail
-
-OT_BUILD_JOBS=$(getconf _NPROCESSORS_ONLN)
-readonly OT_BUILD_JOBS
-
-reset_source()
-{
-    rm -rf build output tmp
-}
-
-build_all_features()
-{
-    local options=(
-        "-DOPENTHREAD_CONFIG_ANNOUNCE_SENDER_ENABLE=1"
-        "-DOPENTHREAD_CONFIG_ANYCAST_LOCATOR=1"
-        "-DOPENTHREAD_CONFIG_BORDER_AGENT_ENABLE=1"
-        "-DOPENTHREAD_CONFIG_BORDER_ROUTER_ENABLE=1"
-        "-DOPENTHREAD_CONFIG_BORDER_ROUTING_ENABLE=1"
-        "-DOPENTHREAD_CONFIG_CHANNEL_MANAGER_ENABLE=1"
-        "-DOPENTHREAD_CONFIG_CHANNEL_MONITOR_ENABLE=1"
-        "-DOPENTHREAD_CONFIG_COAP_API_ENABLE=1"
-        "-DOPENTHREAD_CONFIG_COAP_SECURE_API_ENABLE=1"
-        "-DOPENTHREAD_CONFIG_COMMISSIONER_ENABLE=1"
-        "-DOPENTHREAD_CONFIG_DATASET_UPDATER_ENABLE=1"
-        "-DOPENTHREAD_CONFIG_DHCP6_CLIENT_ENABLE=1"
-        "-DOPENTHREAD_CONFIG_DHCP6_SERVER_ENABLE=1"
-        "-DOPENTHREAD_CONFIG_DIAG_ENABLE=1"
-        "-DOPENTHREAD_CONFIG_DNS_CLIENT_ENABLE=1"
-        "-DOPENTHREAD_CONFIG_DNS_DSO_ENABLE=1"
-        "-DOPENTHREAD_CONFIG_ECDSA_ENABLE=1"
-        "-DOPENTHREAD_CONFIG_HEAP_EXTERNAL_ENABLE=1"
-        "-DOPENTHREAD_CONFIG_HISTORY_TRACKER_ENABLE=1"
-        "-DOPENTHREAD_CONFIG_IP6_FRAGMENTATION_ENABLE=1"
-        "-DOPENTHREAD_CONFIG_IP6_SLAAC_ENABLE=1"
-        "-DOPENTHREAD_CONFIG_JAM_DETECTION_ENABLE=1"
-        "-DOPENTHREAD_CONFIG_JOINER_ENABLE=1"
-        "-DOPENTHREAD_CONFIG_LINK_RAW_ENABLE=1"
-        "-DOPENTHREAD_CONFIG_LOG_LEVEL_DYNAMIC_ENABLE=1"
-        "-DOPENTHREAD_CONFIG_MAC_BEACON_RSP_WHEN_JOINABLE_ENABLE=1"
-        "-DOPENTHREAD_CONFIG_MAC_FILTER_ENABLE=1"
-        "-DOPENTHREAD_CONFIG_MAC_RETRY_SUCCESS_HISTOGRAM_ENABLE=1"
-        "-DOPENTHREAD_CONFIG_MAC_SOFTWARE_ACK_TIMEOUT_ENABLE=1"
-        "-DOPENTHREAD_CONFIG_MAC_SOFTWARE_CSMA_BACKOFF_ENABLE=1"
-        "-DOPENTHREAD_CONFIG_MAC_SOFTWARE_ENERGY_SCAN_ENABLE=1"
-        "-DOPENTHREAD_CONFIG_MAC_SOFTWARE_RETRANSMIT_ENABLE=1"
-        "-DOPENTHREAD_CONFIG_MAC_SOFTWARE_TX_SECURITY_ENABLE=1"
-        "-DOPENTHREAD_CONFIG_MLE_ATTACH_BACKOFF_ENABLE=1"
-        "-DOPENTHREAD_CONFIG_MLE_STEERING_DATA_SET_OOB_ENABLE=1"
-        "-DOPENTHREAD_CONFIG_MPL_DYNAMIC_INTERVAL_ENABLE"
-        "-DOPENTHREAD_CONFIG_NCP_HDLC_ENABLE=1"
-        "-DOPENTHREAD_CONFIG_NCP_SPI_ENABLE=1"
-        "-DOPENTHREAD_CONFIG_PING_SENDER_ENABLE=1"
-        "-DOPENTHREAD_CONFIG_PLATFORM_FLASH_API_ENABLE=1"
-        "-DOPENTHREAD_CONFIG_PLATFORM_RADIO_COEX_ENABLE=1"
-        "-DOPENTHREAD_CONFIG_PLATFORM_USEC_TIMER_ENABLE=1"
-        "-DOPENTHREAD_CONFIG_REFERENCE_DEVICE_ENABLE=1"
-        "-DOPENTHREAD_CONFIG_SNTP_CLIENT_ENABLE=1"
-        "-DOPENTHREAD_CONFIG_SRP_CLIENT_ENABLE=1"
-        "-DOPENTHREAD_CONFIG_TMF_NETDATA_SERVICE_ENABLE=1"
-        "-DOPENTHREAD_CONFIG_TMF_NETDIAG_CLIENT_ENABLE=1"
-        "-DOPENTHREAD_CONFIG_UDP_FORWARD_ENABLE=1"
-        "-DOPENTHREAD_CONFIG_MAC_BEACON_PAYLOAD_PARSING_ENABLE=1"
-        "-DOPENTHREAD_CONFIG_MAC_OUTGOING_BEACON_PAYLOAD_ENABLE=1"
```

Comment:
Do we cover the same set of configs under `cmake` now (e.g. in `build_all_features()`.  e.g. do we have `OT_TX_BEACON_PAYLOAD`  and other configs here?

It is important to cover all features in `build_all_features()` so to make sure a new PR would not break any component.